### PR TITLE
SOLR-15831: Refactor bin/solr and bin/solr.cmd to delegate args parsing and usage help to SolrCLI

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-15831

# Description

This picks up on work that was first done in #504 to migrate logic from shell and command scripts to Java logic.

# Solution

Migrated logic over to Java for:

1) package tool.  PackageTool now uses the same methods (recently added) to Tool for putting out custom usage help.  I did remove the check that you are running packagetool on the local Solr server since the tool works without that requirement.

2) Delete tool.   I pulled in some custom descriptive text from #504 to explain how delete works.  

3) auth tool.  I moved over the argument parsing to the Java code.

# Tests
bats and existing unit tests

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/cli/AuthTool.java [hunk 1]
```diff
@@ -59,6 +60,26 @@ public String getName() {
     return "auth";
   }
 
+  @Override
+  public String getUsage() {
```
Comment (epugh) [inline solr/core/src/java/org/apache/solr/cli/AuthTool.java:64]: @janhoy would you mind looking at the output of the help for this tool?  `bin/solr auth -h`?  Did I implement getUsage and getHeader the way you thought they would be?

File: solr/core/src/java/org/apache/solr/cli/DeleteTool.java [hunk 2]
```diff
@@ -59,6 +59,16 @@ public String getName() {
...
   }
 
+  @Override
+  public String getHeader() {
+    return "\nDeletes a core or collection depending on whether Solr is running in standalone (core) or SolrCloud"
+        + " mode (collection). If you're deleting a collection in SolrCloud mode, the default behavior is to also"
+        + " delete the configuration directory from Zookeeper so long as it is not being used by another collection.\n"
+        + " You can override this behavior by passing -deleteConfig false when running this command.\n"
+        + "\n"
+        + "List of options:";
```
Comment (epugh) [inline solr/core/src/java/org/apache/solr/cli/DeleteTool.java:69]: @janhoy  to get the formatting to work, I had to list "List of options:" here!

Comment (janhoy) [reply solr/core/src/java/org/apache/solr/cli/DeleteTool.java:69]: Yea, guess that's how commons-cli works. We could of course append that "List of options:" string in SolrCLI so tools only need to provide the text itself...

File: solr/core/src/java/org/apache/solr/cli/AuthTool.java (hunk 1 above)
Comment (janhoy) [reply solr/core/src/java/org/apache/solr/cli/AuthTool.java:64]: Looks to be as intended, yea. DId not compile the PR to print the help though.

**Review by janhoy — COMMENTED** [review]
Great to see first usage of the new interface methods...

**Comment by epugh** [comment]
@janhoy  what do you think about me merging this?

**Comment by janhoy** [comment]
Sure, great steps in right direction

**Pull request merged**